api/app/controllers/spree/api/checkouts_controller.rb

```diff
@@ -56,14 +56,7 @@ def complete
       def update
         authorize! :update, @order, order_token
 
-        update_params = if params[:payment_source].present?
-          ActiveSupport::Deprecation.warn("Passing payment_source is deprecated. Send source parameters inside payments_attributes[:source_attributes].", caller)
-          move_payment_source_into_payments_attributes(params)
-        else
-          params
-        end
-
-        if @order.update_from_params(update_params, permitted_checkout_attributes, request.headers.env)
+        if OrderUpdateAttributes.new(@order, update_params, request_env: request.headers.env).apply
```

Comment:
@jhawthorn should this be a configurable class? e.g. `Spree::Config.order_update_attributes_class`